src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -115,6 +118,23 @@ class WorkflowManagerActor(dataAccess: DataAccess, backend: Backend) extends Act
     }
   }
 
+  private def assertCallFqnWellFormed(callFqn: FullyQualifiedName): Future[(String, String)] = {
+    callFqn.split("\\.").toSeq match {
+      case s:Seq[String] if s.size >= 2 => Future.successful(s.head, s.last)
```

Comment:
If I understand you correctly that it at least "works" currently and would need to be changed imminently I'd say to treat fixing it right as a stretch goal (i.e. don't let it block this being completed by end of sprint) and if not to put it as part of the scatter-gather work.